modules/programs/nh.nix

```diff
@@ -0,0 +1,110 @@
+# Based off: https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/programs/nh.nix
+# When making changes please try to keep it in sync.
+{ config
+, lib
+, pkgs
+, ...
+}:
+
+with lib;
+
+let
+  cfg = config.programs.nh;
+in
+{
+  meta.maintainers = [
+    maintainers.alanpearce or "alanpearce"
+  ];
+
+  imports = [
+    (mkRemovedOptionModule [ "programs" "nh" "clean" "dates" ] "Use `programs.nh.clean.interval` instead.")
+  ];
+
+  options.programs.nh = {
+    enable = mkEnableOption "nh, yet another Nix CLI helper";
+
+    package = mkPackageOption pkgs "nh" { };
```

Comment:
Instead could we use https://github.com/ToyVo/nh_darwin instead of the default nh package as it doesn't support `darwin-rebuild` when nh_darwin does!

Perhaps making it a input?

Reply:
That would make sense, but requires nh_darwin to be in nixpkgs. As it is, users will be able to choose to use that package if they have imported it.

Reply:
nh makes errors under darwin - darwin has to use the nh_darwin fork.

For example there are checks on the user id - nh uses the Linux rules and so only checks for ids >= 1000 nh_darwin understands that macOS users start at 501. This affect nh clean as far as I remember

I don't think there is any point in a darwin module that does not use the nh_darwin fork.

Reply:
> That would make sense, but requires nh_darwin to be in nixpkgs. As it is, users will be able to choose to use that package if they have imported it.

I agree with this, @ToyVo Are you willing to make the package nh_darwin in nixpkgs?

Reply:
I mean I could but the problem is there is no ref tags yet until that is a thing I really can't make the package.

CC @ToyVo perhaps make a tag for each update and go based on what nh does already.

Reply:
It does not need to go in nixpkgs.

Just add it to the inputs in your flake.nix (As far as I see nh only deals with flakes so you will have a flake.nix to control your system) See the nix-darwin setup in https://github.com/ToyVo/nh_darwin

Reply:
> It does not need to go in nixpkgs.
> 
> Just add it to the inputs in your flake.nix (As far as I see nh only deals with flakes so you will have a flake.nix to control your system) See the nix-darwin setup in https://github.com/ToyVo/nh_darwin

The only problem for package types all packages are required to be on nixpkgs.

Reply:
I have migrated to nh_plus, I've made the first tag of 3.6.0-0. This puts it in a better spot to bring into nixpkgs

Reply:
I will make the package later today, as I am at school.

Reply:
I didn't see a pr for it so https://github.com/NixOS/nixpkgs/pull/357106

Reply:
Change of plans, with base nh 4.0 nix-darwin support has been added. nh 4.0 is less flake-centric and updating flake inputs has been removed so I might keep my fork alive for that if it isn't re-added.